### PR TITLE
[onert][gpu_cl] Apply shape on enqueueWriteBuffer/enqueueReadBuffer.

### PR DESCRIPTION
**Pull request by Hyunjun85**
Apply shape on enqueueWriteBuffer/enqueueReadBuffer.

1. When the ClBuffer is created, It was aligned by the channel size.
   So While trying to read or write the buffer, It must be aligned.
2. And it will be applied a shape on enqueueWriteBuffer/enqueueReadBuffer.

Signed-off-by: H. Kim hyunjun2.kim@samsung.com
Signed-off-by: hj0412-yi hj0412.yi@samsung.com
Signed-off-by: sungho22.lee sungho22.lee@samsung.com

### CODE REVIEW

**Review by juitem — COMMENTED** [review]
would you please put [gpu_cl] after one on title?

**Comment by chunseoklee** [comment]
@Hyunjun85 Could you create DRAFT PR to show the final structure of this PR ? (better to have test case, too)

**Comment by Hyunjun85** [comment]
> would you please put [gpu_cl] after one on title?

Applied.

File: runtime/onert/backend/gpu_cl/operand/ICLTensor.cc [hunk 1]
```diff
@@ -37,55 +37,60 @@ void ICLTensor::access(const std::function<void(ITensor &tensor)> &fn)
...
+      dst_tensor.shape = BHWC(_shape.dim(0), 1, 1, _shape.dim(1));
       break;
-    default:
+    case 3:
+      dst_tensor.shape = BHWC(_shape.dim(0), 1, _shape.dim(1), _shape.dim(2));
+      break;
+    case 4:
+      dst_tensor.shape = BHWC(_shape.dim(0), _shape.dim(1), _shape.dim(2), _shape.dim(3));
       break;
   }
```
Comment (juitem) [inline runtime/onert/backend/gpu_cl/operand/ICLTensor.cc:85]: It seems limited condition, would you please check newly applied

**Pull request closed without merging**